test/metabase/models/collection_test.clj

```diff
@@ -1599,25 +1599,31 @@
         (is (= [{:id       3
                  :name     "a"
                  :location "/"
-                 :children [{:id       1
-                             :name     "a"
-                             :location "/3/"
-                             :children [{:id       2
-                                         :name     "a"
-                                         :location "/3/1/"
-                                         :children [{:id       5
-                                                     :name     "a"
-                                                     :location "/3/1/2/"
-                                                     :children []}]}
-                                        {:id       4
-                                         :name     "a"
-                                         :location "/3/1/"
-                                         :children []}]}
-                            {:id       6
-                             :name     "a"
-                             :location "/3/"
-                             :children []}]}]
-               (collection/collections->tree {} collections)))))))
+                 :children #{{:id       1
+                              :name     "a"
+                              :location "/3/"
+                              :children #{{:id       2
+                                           :name     "a"
+                                           :location "/3/1/"
+                                           :children #{{:id       5
+                                                        :name     "a"
+                                                        :location "/3/1/2/"
+                                                        :children #{}}}}
+                                          {:id       4
+                                           :name     "a"
+                                           :location "/3/1/"
+                                           :children #{}}}}
+                             {:id       6
+                              :name     "a"
+                              :location "/3/"
+                              :children #{}}}}]
+               (walk/postwalk (fn [m]
+                                (if (and (map? m) (:children m))
+                                  (update m :children set)
```

Comment:
This is kind of a weird test, it's doing `math/permutations` to put those 6 collections into every possible order, then asserts that the result is the same every time. I did some digging and it was added because there was a bug where we generated a bad tree when the collections were passed in a particular order.

After my changes, the caller is now responsible for sorting the collections passed in, and `collections->tree` will return them in the same order. So while we still want this test to be sure that we're not generating a completely invalid tree for some permutation, it's ok as long as we generate a valid tree that represents the collections we passed in.

But that does make me realize we don't actually have a test for *that*... I'll add that.